File: Android/SocketServer/src/main/RawMessage.java
package main;

public class RawMessage {
    int headerLen, fileLen;
    byte[] rawData;

    public RawMessage(int headerLen, int fileLen, byte[] rawData) {
        this.headerLen = headerLen;
        this.fileLen = fileLen;
        this.rawData = rawData;
    }

    public RawMessage(byte[] rawData) {
        this.headerLen = -1;
        this.fileLen = -1;
        this.rawData = rawData;
    }

    public int getHeaderLen() {
        return headerLen;
    }

    public void setHeaderLen(int headerLen) {
        this.headerLen = headerLen;
    }

    public int getFileLen() {
        return fileLen;
    }

    public void setFileLen(int fileLen) {
        this.fileLen = fileLen;
    }

    public byte[] getRawData() {
        return rawData;
    }

    public void setRawData(byte[] rawData) {
        this.rawData = rawData;
    }
}

File: Android/SSL_Chat/mynetworklibrary/src/main/cpp/chat/RawMessage.cpp
//
// Created by Ivan Yurovych on 1/23/19.
//

#include "RawMessage.h"

RawMessage::RawMessage(int headerLen, int fileLen, bool isFromServer, const std::string &data)
        : headerLen(headerLen), fileLen(fileLen), fromServer(isFromServer), data(data) {}

int RawMessage::getHeaderLen() const {
    return headerLen;
}

int RawMessage::getFileLen() const {
    return fileLen;
}

bool RawMessage::isFromServer() const {
    return fromServer;
}

const std::string &RawMessage::getData() const {
    return data;
}
